drivers/sip_svc/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2021, Intel Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_library()
+
+if(CONFIG_ARM_SIP_SVC)
+  zephyr_library_sources(sip_svc_ll.c)
+  zephyr_library_sources(sip_svc_ll_id_mgr.c)
+  zephyr_library_sources(sip_svc.c)
```

Comment:
`CONFIG_ARM_SIP_SVC` is supposed to represent a class of drivers (for example `CONFIG_LED`, `CONFIG_CAN`, `CONFIG_AUDIO`, etc...) but here you are unconditionally compiling a very specific low-level implementation.

Reply:
The "sip_svc/" folder will only be included ifdef(CONFIG_ARM_SIP_SVC) in drivers/CMakeList.txt. So, I thought drivers/sip_svc/CMakeLists.txt will only be compiled ifdef(CONFIG_ARM_SIP_SVC). Thus, I don't have other condition check for these 3 files because they are needed ifdef(CONFIG_ARM_SIP_SVC).

Kindly correct me if I am not doing in right way.